### DISCUSSION TITLE
Is there a way to display the xstate inspector in an iframe that is not immediately available on the DOM?

### DISCUSSION THREAD

**Discussion by WilliamIPark** [statelyai/xstate#3470] [Q&A]
Hello, 

I'm currently building a set of dev-tools for our Next app, which lives in a popover panel. One of the tools I wish to move into here is the xstate inspector, but I've been having trouble getting to it to appear and work correctly.

It appears that the inspector needs called as early as possible in your app, which means I can't just call it during the React lifecycle, but I need a way to target an iframe that won't exist until someone opens the devtools panel. 

I've seen you can eagerly or lazily specify the iframe property, but doing neither helps this case unfortunately!

**Comment by Andarist** [comment]
We will be improving this in the upcoming weeks.

**Reply by WilliamIPark** [reply]
Ah brilliant, that's great news! Is there an issue for this I can keep an eye on?

**Reply by Andarist** [reply]
I imagine that we will try to work on this while preparing the browser extension for XState, so I would stay tuned for that release.
